packages/contracts/contracts/optimistic-ethereum/ovm/SafetyChecker.sol

```diff
@@ -63,136 +35,102 @@ contract SafetyChecker is ContractResolver {
-                    // encountered a non-whitelisted opcode!
-                    return false;
-                }
-                // append this opcode to a list of ops
-                ops <<= 8;
-                ops |= op;
-                // [0x57, 0x56, 0x00, 0xfd, 0xfe, 0xf3, 0xf1] all have handlers
-                if (opBit & 0x600a00000000000000000000000000000000000000c000000000000000000001 != 0) {
-                    // JUMPI
-                    if (op == 0x57) {
-                        // We can now reach all JUMPDESTs
-                        seenJUMP = true;
-                    // JUMP
-                    } else if (op == 0x56) {
-                        // We can now reach all JUMPDESTs
-                        seenJUMP = true;
-                        // we are now inside unreachable code until we hit a JUMPDEST!
-                        insideUnreachableCode = true;
-                    // STOP or REVERT or INVALID or RETURN (see safety checker docs in wiki for more info)
-                    } else if (op == 0x00 || op == 0xfd || op == 0xfe || op == 0xf3) {
-                        // If we can't jump to JUMPDESTs, then all remaining bytecode is unreachable
-                        if (!seenJUMP) {
-                            return true;
-                        }
-                        // We are now inside unreachable code until we hit a JUMPDEST!
-                        insideUnreachableCode = true;
-                    // CALL
-                    } else if (op == 0xf1) {
-                        // Minimum 4 total ops:
-                        // 1. PUSH1 value
-                        // 2. PUSH20 execution manager address
-                        // 3. PUSH or DUP gas
-                        // 4. CALL
+            // inline assembly removes the extra add + bounds check
+            assembly {
+                let tmp := mload(_pc)
 
-                        // if opIndex < 3, there would be 0s here, so we don't need the check
-                        uint256 gasOp = (ops >> 8) & 0xFF;
-                        uint256 addressOp = (ops >> 16) & 0xFF;
-                        uint256 valueOp = (ops >> 24) & 0xFF;
-                        if (
-                            gasOp < 0x60 || // PUSHes are 0x60...0x7f
-                            gasOp > 0x8f || // DUPs are 0x80...0x8f
-                            addressOp != 0x73 || // address must be set with a PUSH20
-                            valueOp != 0x60 // value must be set with a PUSH1
-                        ) {
-                            return false;
-                        } else {
-                            uint256 pushedBytes;
-                            // gas is set with a PUSH##
-                            if (gasOp >= 0x60 && gasOp <= 0x7f) {
-                                pushedBytes = gasOp - 0x5f;
-                            }
+                let mpc := byte(0, mload(add(skip, byte(0, tmp))))
```

Comment:
`mpc` seems like it stands for something. Would rename it to be something that indicates what that is.